src/shared-components/dropdown/index.tsx

```diff
@@ -31,21 +37,21 @@ type DropdownProps = {
   /**
    * The currently selected option
    */
-  value?: string | number;
+  value?: OptionValue;
 };
 
 /**
  * `<Dropdown />` is a controlled component and should be wrapped by a parent to control the dropdown's state.
  * This ships with a mobile implementation that will handle mobile devices automatically.
  */
-export const Dropdown = ({
+export const Dropdown = <T extends OptionType>({
```

Comment:
The majority of changes are to support using Generics to infer the type in `onChange`, based on the `options` provided. Here is a screenshot that shows the behavior:

<img width="547" alt="Screen Shot 2020-11-04 at 2 27 43 PM" src="https://user-images.githubusercontent.com/13544620/98166903-ce0bfe00-1ead-11eb-9995-39ee05e6ed96.png">

And the simpler case that applies to most of our usage:

<img width="549" alt="Screen Shot 2020-11-04 at 2 56 04 PM" src="https://user-images.githubusercontent.com/13544620/98166958-e8de7280-1ead-11eb-94f4-86d1eea85c6c.png">

This gets around the problem we have with `semantic-ui-react` usage where the type is flexible and so we often have to cast things `as string` and such in order to comply with TypeScript. No longer! It will be determined automatically by the options we pass in 🙂